modules/contributor-guide/partials/proc_adding-a-che-theia-plug-in-into-a-workspace.adoc

```diff
@@ -7,67 +7,64 @@
 
 You can add Che-Theia plug-in into the Che-Theia IDE in three ways:
 
-* Using the Che-Theia dashboard
-* By configuring a workspace
-* At runtime
+* At Theia runtime
+* By configuring a workspace devfile
+* By preparing shareable Devfile
 
+[id="adding-a-plug-in-at-runtime_{context}"]
+== Adding a plug-in at Theia runtime
 
-[id="adding-a-plug-in-through-the-dashboard_{context}"]
-== Adding a plug-in through the dashboard
+This method adds a plug-in to an already running workspace. The plug-in is not added permanently; it is removed when the workspace stops. This is a good method to test a plug-in without affecting the permanent workspace configuration.
 
-. Navigate to the *Workspaces* tab in the dashboard.
-. Select the workspace to which you want to add the plug-in.
-. In *Workspace > _<workspace-name>_* window, click the *Plugins* tab, and select the required plug-ins.
+. Run the `Deploy Plugin by Id` command, and specify the ID of the plug-in that you want to add to the workspace. A plug-in ID can be:
++
+** A local directory with plug-ins. All the plug-in binaries (`.theia` files) in the specified directory would be loaded. Example: `local-dir:///home/theia/plugins/`.
+
+** VS Code extension link. To obtain the link, go to the VS Code marketplace, find the needed extension, copy its ID, and append the ID to the `vscode:extension/` prefix.
 +
-.Adding plug-ins through the dashboard
-image::extensibility/adding-plug-ins-in-dashboard.png[link="../_images/extensibility/adding-plug-ins-in-dashboard.png"]
+For example, for the link:https://marketplace.visualstudio.com/items?itemName=redhat.vscode-xml[XML language server] plug-in, it is: `vscode:extension/redhat.vscode-xml`.
 
+** Link to plug-in binaries (`.theia` files). Can be both HTTP and HTTPS. Example: `https://domain.net/path/plugin.theia`.
```

Comment:
These lines are made obsolete by the previous suggestion, but GitHub UI doesn't handle well multi-lines suggestions when deleted lines are involved. 
```suggestion
```

Reply:
Removed